### PR TITLE
Upgraded to Python 3.7

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Updated the Python version of ark-analysis to 3.7.

**How did you implement your changes**

Had to adjust some Pandas and xarray indexing. Had to update the following packages:

**Requirements**

| Package  | Python 3.6 Version | Python 3.7 Version |
| ------------- | ------------- | ------------- |
| jupyterlab  | 3.1.5  | 3.4.3  |
| numpy  | 1.16.3  | 1.21.6  |
| pandas  | 0.23.3  | 1.3.5  |
| requests  | 2.25.1  | 2.27.6  |
| scikit-image  | 1.14.3  | 1.16.2  |
| scikit-learn  | 0.19.1  | 0.24.2  |
| scipy  | 1.1.0  | 1.7.3  |
| seaborn  | 0.10.1  | 0.11.2  |
| statsmodels  | 0.11.1  | 0.13.2 |
| tables  | 3.6.1 | 3.7.0  |
| umap-learn  | 0.4.6  | 0.5.3  |
| xarray  | 0.12.3  | 0.17.0  |
| tqdm  | 4.54.1  | 4.64.0  |

**Requirements Test**

| Package  | Python 3.6 Version | Python 3.7 Version |
| ------------- | ------------- | ------------- |
| attrs  | 20.3.0  | 21.4.0  |
| coveralls  | 1.8.2  | 1.11.1  |
| pytest  | 6.0.0  | 7.1.2  |
| pytest-asyncio  | 0.15.1  | 0.18.1  |
| six  | 1.13.0  | 1.16.0  |
| testbook  | 0.2.3  | 0.4.2  |
| coveralls  | 1.8.2  | 1.11.1  |

**Travis CI**

Added a line to install the latest versions of `setuptools` and `importlib-metadata`. Travis-CI Python 3.7 environments contain a very old version of `importlib-metadata` which is incompatible with the majority of libraries for 3.7+.

**Remaining issues**

[ ] Docker builds successfully

**Updating other libraries**

In order for ark-analysis to work with toffy, both toffy and mibi-bin-tools need a couple libraries to be updated.

Order of Updates:

1. Merge angelolab/ark-analysis#592 (this one)
2. Merge angelolab/mibi-bin-tools#36
3. Merge angelolab/toffy#132

### CODE REVIEW

File: docs/_rtd/development.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ To test if `conda` has been added to your path, run `conda info` in your Termina
 
 Now that Anaconda is installed, you can now create a `conda` environment. 
  
-To do so, on your command line, type `conda create -n <my_env> python=3.6`, where `<my_env>` is a name you set. Our codebase only supports Python 3.6, so please do not change the `python=3.6` flag when creating your environment. 
+To do so, on your command line, type `conda create -n <my_env> python=3.7`, where `<my_env>` is a name you set. Our codebase only supports Python 3.6, so please do not change the `python=3.7` flag when creating your environment.
```
Comment (ngreenwald) [inline docs/_rtd/development.md:23]:
```suggestion
To do so, on your command line, type `conda create -n <my_env> python=3.7`, where `<my_env>` is a name you set. Our codebase only supports Python 3.7, so please do not change the `python=3.7` flag when creating your environment. 
```

File: requirements.txt [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-git+https://github.com/calico/spatial_lda.git@primary
-statsmodels>=0.11.1,<1
-tables>=3.6.1,<4
-umap-learn>=0.4.6,<1
-xarray>=0.12.3,<1
-tqdm>=4.54.1,<5
+scikit-learn>=0.24.2,<1
+scipy>=1.7.3,<2
+seaborn>=0.11.2,<1
+spatial-lda>=0.1.3,<1
```
Comment (ngreenwald) [inline requirements.txt:18]: Can you check to confirm that this release of spatial-lda isn't significantly behind the github? They were slow to make a release before, which is why we were using the github

File: setup.py [hunk 3]
```diff
@@ -21,24 +21,24 @@
...
-                      'tqdm>=4.54.1,<5'],
-    dependency_links=[
-        'git+https://github.com/calico/spatial_lda.git@primary',
-    ],
+                      'statsmodels>=0.13.2,<1',
+                      'umap-learn>=0.5.3,<1',
+                      'xarray>=0.17.0,<1',
+                      'tqdm>=4.64.0,<5'],
+    # dependency_links=[
+    #     'git+https://github.com/calico/spatial_lda.git@primary',
```
Comment (ngreenwald) [inline setup.py:40]: If it's up to date, we can remove this (I think?). If not we'll keep everything linked to the github

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Looks good, just a couple changes

File: requirements.txt (hunk 2 above)
Comment (srivarra) [reply requirements.txt:18]: I confirmed it with Adam a few days ago. He stated that 0.1.3 good to use.

File: setup.py (hunk 3 above)
Comment (srivarra) [reply setup.py:40]: Yes, this indeed can be removed.

File: setup.py [hunk 4]
```diff
@@ -21,24 +21,21 @@
...
-                      'umap-learn>=0.4.6,<1',
-                      'xarray>=0.12.3,<1',
-                      'tqdm>=4.54.1,<5'],
-    dependency_links=[
-        'git+https://github.com/calico/spatial_lda.git@primary',
-    ],
+                      'statsmodels>=0.13.2,<1',
+                      'umap-learn>=0.5.3,<1',
+                      'xarray>=0.17.0,<1',
+                      'tqdm>=4.64.0,<5'],
```
[the remark this answers is not in the document]
Comment (srivarra) [reply setup.py:38]: Yes, it should be possible to use that method.

**Comment by srivarra** [comment]
@alex-l-kong Yeah, I tested it with toffy, there are a couple of changes we need to make to both mibi-bin-tools and toffy w.r.t. updating libraries. 

See angelolab/toffy#132, angelolab/mibi-bin-tools#36.

**Review by alex-l-kong — COMMENTED** [review]
Looks good. We'll need to do a tagged commit to see if PyPI deploys properly and the Docker image builds on Dockerhub.

**Comment by ngreenwald** [comment]
Once you guys are satisfied that current docker works well enough on M1, let me know and we can merge it in. I don't think we need to resolve #611 before moving forward, can add it to the todolist

**Comment by srivarra** [comment]
@ngreenwald This is good to go, everything works well.

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**